src/Domain/Syntax/DistanceFeature.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JeroenG\Explorer\Domain\Syntax;
+
+use JeroenG\Explorer\Domain\Syntax\SyntaxInterface;
+
+class DistanceFeature implements SyntaxInterface
+{
+    private string $field;
+
+    private ?float $boost;
+
+    private mixed $pivot;
+
+    private mixed $origin;
```

Comment:
Origin is a string if I read the docs correctly.

Reply:
And geopoint is an array in that case?

Reply:
Ok wow, no I thought it could be either string or array, and in PHP8 you can do a `string|array` typehint AFAIK.